tgui/packages/tgui/interfaces/Uplink/PrimaryObjectiveMenu.tsx

```diff
@@ -0,0 +1,78 @@
+import { Box, Dimmer, Section } from '../../components';
+import { ObjectiveElement } from './ObjectiveMenu';
+
+type PrimaryObjectiveMenuProps = {
+  primary_objectives;
+  final_objective;
+};
+
+export const PrimaryObjectiveMenu = (
+  props: PrimaryObjectiveMenuProps,
+  context
+) => {
+  const { primary_objectives, final_objective } = props;
+  return (
+    <Section fill>
+      <Section>
+        <Box mt={3} mb={3} bold fontSize={1.2} align="center" color="white">
+          {
+            'Agent, your Primary Objectives are as follows. Complete these at all costs.'
+          }
+        </Box>
+        <Box mt={3} mb={5} bold fontSize={1.2} align="center" color="white">
+          {
+            'Completing Secondary Objectives may allow you to aquire additional equipment.'
+          }
+        </Box>
+      </Section>
+      {final_objective && (
+        <Dimmer>
+          <Box
+            color="red"
+            fontFamily={'Bahnschrift'}
+            fontSize={3}
+            align={'top'}
+            as="span">
+            PRIORITY MESSAGE
+            <br />
+            SOURCE: xxx.xxx.xxx.224:41394
+            <br />
+            <br />
+            \\Debrief in progress.
+            <br />
+            \\Final Objective confirmed complete. <br />
+            \\Your work is done here, agent.
+            <br />
+            <br />
+            CONNECTION CLOSED_
+          </Box>
+        </Dimmer>
+      )}
+      <Section>
+        {primary_objectives.map((prim_obj, index) => (
+          <ObjectiveElement
+            key={prim_obj.id}
+            name={'Objective ' + (index + 1)}
```

Comment:
Calling them Objective 1/2/3 is a bit bland. Could you go for some different naming theme, e.g. 
"First Directive", "Second Directive" or something along those lines.